yjit/src/invariants.rs

```diff
@@ -4,62 +4,89 @@
 use crate::core::*;
 use crate::cruby::*;
 use crate::codegen::*;
-use crate::yjit::{yjit_enabled_p};
+use crate::stats::*;
+use crate::yjit::yjit_enabled_p;
+use std::collections::HashMap;
 
 // Invariants to track:
 // assume_bop_not_redefined(jit, INTEGER_REDEFINED_OP_FLAG, BOP_PLUS)
 // assume_method_lookup_stable(comptime_recv_klass, cme, jit);
 // assume_single_ractor_mode(jit)
 // assume_stable_global_constant_state(jit);
 
+/// A type alias for the redefinition flags coming from CRuby. These are just
+/// shifted 1s but not explicitly an enum.
+type RedefinitionFlag = u32;
```

Comment:
This should probably move to `cruby.rs` and be used in the signature of `pub fn BASIC_OP_UNREDEFINED_P(bop: ruby_basic_operators, klass: u32) -> bool;` and `static inline void rb_yjit_bop_redefined(int redefined_flag, enum ruby_basic_operators bop) {}`

Reply:
Updated!